### ISSUE TITLE
[Bug] Packed FP4 ND tensors use logical width as the GM row stride on A5

### ISSUE THREAD

**Issue by high-cloud** [hw-native-sys/pypto#2754]
## Background

In pypto-lib, a two-row packed FP4 cache tensor on A5 has a precision problem:
row 1 is addressed with a 512-byte step for 512 logical FP4 elements, while its
`torch.float4_e2m1fn_x2` carrier row contains 256 bytes.

Reproduce with:

    python tmp_fp4_stride_repro.py -p a5 -d 0

The reproduction case is not committed. Save the source below as
`tmp_fp4_stride_repro.py` in a public pypto-lib checkout.

Reproduction environment:

| Component | Version |
|---|---|
| pypto-lib | `c3f0dea274f55d9648920f17c968e8564ae9fcdc` (branch: `main`) |
| pypto | `ddf42145191aa8c8fcffc563d2ee644364c22f28` (branch: `main`) |
| simpler | `39ce891dbb3f665e72e99b4a1387b47012fb18bd` (detached) |
| ptoas | `0.61` |
| pto-isa | `5a4f74cbf627d4aac2e0ce10d5e0d8b118343265` (detached) |
| CANN | `9.2.0` |

All installed components match the versions pinned by the current PyPTO
checkout.

Diagnosis: **pypto Codegen**. The JIT boundary correctly expands the physical
x2 carrier shape to 512 logical FP4 elements, and the generated Vec tile
correctly has 256 carrier columns. However, tensor-view lowering emits a GM row
stride and row offset of 512 carrier elements.

### Component

Codegen

### Description

A contiguous ND tensor declared as `pl.Tensor[[2, 512], pl.FP4]` is backed by a
Torch tensor with physical shape `[2, 256]` and dtype
`torch.float4_e2m1fn_x2`. A simple per-row load/store copies row 0 correctly but
reads and writes row 1 at the wrong address.

This prevents packed FP4 cache tensors from using their production PyPTO
signatures. pypto-lib currently has to expose them as
`pl.UINT8[..., logical_width // 2]` and manually pack/unpack E2M1 nibbles.

Related: #2087.

### Steps to Reproduce

<details>
<summary>Reproduction case: <code>tmp_fp4_stride_repro.py</code></summary>

```python
import argparse

import pypto.language as pl
import torch

from golden import TensorSpec, run


ROWS = 2
LOGICAL_WIDTH = 512
PHYSICAL_WIDTH = LOGICAL_WIDTH // 2


@pl.jit
def fp4_copy(
    source: pl.Tensor[[ROWS, LOGICAL_WIDTH], pl.FP4],
    output: pl.Out[pl.Tensor[[ROWS, LOGICAL_WIDTH], pl.FP4]],
):
    for row in pl.spmd(ROWS):
        value = pl.load(source, [row, 0], [1, LOGICAL_WIDTH])
        pl.store(value, [row, 0], output)
    return output


def values():
    physical = torch.empty((ROWS, PHYSICAL_WIDTH), dtype=torch.uint8)
    physical[0] = torch.arange(PHYSICAL_WIDTH, dtype=torch.uint8)
    physical[1] = torch.arange(PHYSICAL_WIDTH, dtype=torch.uint8).bitwise_xor(0xA5)
    return physical.view(torch.float4_e2m1fn_x2)


def golden_copy(tensors):
    tensors["output"].view(torch.uint8).copy_(tensors["source"].view(torch.uint8))


def compare_bytes(actual, expected, **_kwargs):
    actual_bytes = actual.view(torch.uint8)
    expected_bytes = expected.view(torch.uint8)
    mismatches = torch.where(actual_bytes != expected_bytes)
    if not mismatches[0].numel():
        return True, ""
    first = tuple(index[0].item() for index in mismatches)
    return False, (
        f"first byte mismatch at {first}: "
        f"actual={actual_bytes[first]}, expected={expected_bytes[first]}"
    )


if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("-p", "--platform", default="a5")
    parser.add_argument("-d", "--device", type=int, required=True)
    args = parser.parse_args()
    source = values()
    specs = [
        TensorSpec("source", list(source.shape), source.dtype, init_value=source),
        TensorSpec("output", list(source.shape), source.dtype),
    ]
    result = run(
        fp4_copy,
        specs,
        golden_fn=golden_copy,
        compare_fn={"output": compare_bytes},
        config=dict(platform=args.platform, device_id=args.device),
        rtol=0.0,
        atol=0.0,
    )
    if result.error:
        print(result.error)
    raise SystemExit(0 if result.passed else 1)
```

</details>

Run on a real A5 through the normal device allocation mechanism.

### Expected Behavior

Both physical rows are copied byte-for-byte. For logical width 512, advancing
one row must advance 256 `float4_e2m1fn_x2` carrier elements.

### Actual Behavior

The first mismatch is the first byte of row 1:

```text
first byte mismatch at (1, 0): actual=0, expected=165
```

Generated PTO/C++ uses logical width as the carrier stride:

```text
shape = [2, 512], strides = [512, 1]
Tile<..., float4_e2m1x2_t, 1, 512, ...> physical cols = 256
GlobalTensor<float4_e2m1x2_t, ..., Stride<512, 512, 512, 512, 1>>
row pointer offset = row * 512
```

### Git Commit ID

`ddf42145191aa8c8fcffc563d2ee644364c22f28`

### NPU Kind

Other: Ascend A5

### Host Platform

Linux (aarch64)

### Additional Context

Python `3.11.16`, Torch `2.14.0+cpu`.

A regression test should cover at least two rows of a contiguous ND FP4 tensor
and compare its underlying x2 carrier bytes. The fix must keep logical FP4
shapes in IR while converting non-last-axis strides and offsets to carrier
units before pointer arithmetic.
